presto-iceberg/src/main/java/com/facebook/presto/iceberg/ExpressionConverter.java

```diff
@@ -0,0 +1,203 @@
+import com.facebook.presto.common.type.ArrayType;
+import com.facebook.presto.common.type.DateType;
+import com.facebook.presto.common.type.DecimalType;
+import com.facebook.presto.common.type.Decimals;
+import com.facebook.presto.common.type.IntegerType;
+import com.facebook.presto.common.type.MapType;
+import com.facebook.presto.common.type.RealType;
+import com.facebook.presto.common.type.RowType;
+import com.facebook.presto.common.type.Type;
+import com.facebook.presto.common.type.VarbinaryType;
+import com.facebook.presto.common.type.VarcharType;
+import com.google.common.base.VerifyException;
+import io.airlift.slice.Slice;
+import org.apache.iceberg.expressions.Expression;
+
+import java.math.BigDecimal;
+import java.nio.ByteBuffer;
+import java.util.List;
+import java.util.Map;
+
+import static com.facebook.presto.common.predicate.Marker.Bound.ABOVE;
+import static com.facebook.presto.common.predicate.Marker.Bound.BELOW;
+import static com.facebook.presto.common.predicate.Marker.Bound.EXACTLY;
+import static com.google.common.base.MoreObjects.firstNonNull;
+import static com.google.common.base.Preconditions.checkArgument;
+import static java.lang.Float.intBitsToFloat;
+import static java.lang.Math.toIntExact;
+import static java.util.Objects.requireNonNull;
+import static org.apache.iceberg.expressions.Expressions.alwaysFalse;
+import static org.apache.iceberg.expressions.Expressions.alwaysTrue;
+import static org.apache.iceberg.expressions.Expressions.and;
+import static org.apache.iceberg.expressions.Expressions.equal;
+import static org.apache.iceberg.expressions.Expressions.greaterThan;
+import static org.apache.iceberg.expressions.Expressions.greaterThanOrEqual;
+import static org.apache.iceberg.expressions.Expressions.isNull;
+import static org.apache.iceberg.expressions.Expressions.lessThan;
+import static org.apache.iceberg.expressions.Expressions.lessThanOrEqual;
+import static org.apache.iceberg.expressions.Expressions.not;
+import static org.apache.iceberg.expressions.Expressions.or;
+
+public final class ExpressionConverter
+{
+    private ExpressionConverter() {}
+
+    public static Expression toIcebergExpression(TupleDomain<IcebergColumnHandle> tupleDomain)
+    {
+        if (tupleDomain.isAll()) {
+            return alwaysTrue();
+        }
+        if (!tupleDomain.getDomains().isPresent()) {
+            return alwaysFalse();
+        }
+        Map<IcebergColumnHandle, Domain> domainMap = tupleDomain.getDomains().get();
+        Expression expression = alwaysTrue();
+        for (Map.Entry<IcebergColumnHandle, Domain> entry : domainMap.entrySet()) {
+            IcebergColumnHandle columnHandle = entry.getKey();
+            Domain domain = entry.getValue();
+            expression = and(expression, toIcebergExpression(columnHandle.getName(), columnHandle.getType(), domain));
+        }
+        return expression;
+    }
+
+    private static Expression toIcebergExpression(String columnName, Type type, Domain domain)
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 24. By comparison, 99% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test. We recommend that you simplify this method or break it into multiple methods.